File: patterns/java/src/refactoring_guru/patterns/singleton/example/non_thread_safe/Singleton.java
package refactoring_guru.patterns.singleton.example.non_thread_safe;

public final class Singleton {
    private static Singleton instance;
    public String value;

    private Singleton(String value) {
        // Emulates time consuming operation.
        try {
            Thread.currentThread().sleep(1000);
        } catch (InterruptedException ex) {
            ex.printStackTrace();
        }
        this.value = value;
    }

    public static Singleton getInstance(String value) {
        if (instance == null) {
            instance = new Singleton(value);
        }
        return instance;
    }
}

File: patterns/cs/src/RefactoringGuru/NonThreadSafeSingleton/Example/Singleton.cs
using System;

namespace NonThreadSafeSingleton.Example
{
    class Singleton
    {
        private static Singleton instance;
        public String value;

        private Singleton(String value)
        {
            // EN: Following code emulates slow initialization.
            // 
            // RU: Этот код эмулирует медленную инициализацию.
            System.Threading.Thread.Sleep(50);
            this.value = value;
        }

        public static Singleton GetInstance(String value)
        {
            if (instance == null)
            {
                instance = new Singleton(value);
            }
            return instance;
        }
    }
}
